.github/workflows/codeql-analysis.yml

```diff
@@ -0,0 +1,26 @@
+name: "CodeQL"
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
```

Comment:
It does run on every pull request.

Reply:
I see.  The `branches` specifies which branch(es) the pull request is against.